index.bs

```diff
@@ -78,4 +251,313 @@ interface StorageBucket {
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">durability()</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+    1. If |bucket| has been removed, [=reject=] |p| with an {{InvalidStateError}}.
+
+    1. Otherwise, [=/resolve=] |p| with |bucket|'s [=bucket durability=].
+
+1. Return |p|.
+
+</div>
+
+<aside class="note">
+
+The durability reflects whether the user agent will prioritize performance or durability when completing operations on data in the bucket.
+Some operations, such as {{IDBTransaction}}s, can specify a durability which will override the [=bucket durability=]. The default
+[=bucket durability=] is left to the user agent. Durability can be specified when a bucket is initially created, but not changed later.
+The {{StorageBucket/durability()}} method should report what the user agent will actually do. The value is one of the following:
+
+: "{{StorageBucketDurability/strict}}"
+:: The user agent will consider that operations on data in the bucket are successful only after verifying that outstanding changes have been written to a persistent storage medium.
+: "{{StorageBucketDurability/relaxed}}"
+:: The user agent will consider that operations on data in the bucket are successful as soon as all outstanding changes have been written to the operating system, without subsequent verification.
+
+</aside>
+
+<h3 id="storage-bucket-expiration">Expiration</h3>
+
+A [=/storage bucket=] has a <dfn>bucket expiration</dfn>, a timestamp-or-null, initially null.
+Specifies the upper limit of a bucket lifetime.
+
+User agents MUST remove any bucket that has expired when {{StorageBucketManager/keys}} is called.
+User agents MUST remove a bucket that has expired when it is opened via {{StorageBucketManager/open()}}.
+User agents MAY clear buckets whose [=/bucket mode=] is "<code>best-effort</code>" before the
+specified timestamp when faced with storage pressure.
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">setExpires(|expires|)</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+    1. If |bucket| has been removed, [=reject=] |p| with an {{InvalidStateError}}.
+
+    1. Otherwise, set |bucket|'s [=bucket expiration=] to |expires| and [=/resolve=] |p|.
+
+1. Return |p|.
+
+</div>
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">expires()</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+    1. If |bucket| has been removed, [=reject=] |p| with an {{InvalidStateError}}.
+
+    1. Otherwise, [=/resolve=] |p| with |bucket|'s [=bucket expiration=].
+
+1. Return |p|.
+
+</div>
+
+<h3 id="storage-bucket-indexeddb">Using Indexed Database</h3>
```

Comment:
It might be helpful to the reader to group these under a section that says "Now we define all the storage APIs that are supported within a bucket" or something.

Reply:
done